### PR TITLE
Source Google Ads: handle refresh error

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolve https://github.com/airbytehq/oncall/issues/1872

## How
Handle Refresh errors as config errors

## Recommended reading order
2. `y.python`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py [hunk 1]
```diff
@@ -44,9 +47,17 @@ def __init__(self, credentials: MutableMapping[str, Any]):
         # `google-ads` library version `14.0.0` and higher requires an additional required parameter `use_proto_plus`.
         # More details can be found here: https://developers.google.com/google-ads/api/docs/client-libs/python/protobuf-messages
         credentials["use_proto_plus"] = True
-        self.client = GoogleAdsClient.load_from_dict(credentials, version=API_VERSION)
+        self.client = self.get_google_ads_client(credentials)
         self.ga_service = self.client.get_service("GoogleAdsService")
 
+    @staticmethod
+    def get_google_ads_client(credentials) -> GoogleAdsClient:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py:54]: can we add unit tests for this one?

**Review by lazebnyi — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py:54]: tests added

**Pull request merged**